### DISCUSSION TITLE
VoxCeleb recipe

### DISCUSSION THREAD

**Discussion by desh2608** [lhotse-speech/lhotse#471] [Ideas]
I am working on a data preparation recipe for VoxCeleb. The VoxCeleb corpus is used for speaker verification tasks and as such, comes with a standard list of trial pairs. For my own purpose, I am not preparing it for a SV task so I don't need these pairs, but it is reasonable to assume that a number of people would need it. I am not sure how to store the trials list in the manifests, since it is not exactly a "supervision" or a "recording". Any ideas?

**Comment by pzelasko** [comment]
Yeah that's a good question. I have two ideas:

1) Leave trials as-is and let the users handle it inside speaker ID dataset

2) As part of data prep recipe, for each dev / test split, create two cut-sets where cuts have ids like `trial1`, `trial2`, and so on, and contain the original recordings + supervisions. The users can use `CutPairsSampler` to get trial pairs of cuts for evaluation (note: the dataset has to handle getting a tuple of CutSet as batch).

**Reply by desh2608** [reply]
The trials file also contains a binary value (0/1) indicating whether the trial target is true or false. What would be a clean way to store these in the associated `CutSet` pairs --- perhaps using the `custom` field (https://github.com/lhotse-speech/lhotse/pull/459)?

**Reply by pzelasko** [reply]
Probably the cleanest way is to save two cut sets — one for positive trials and the other for negative. I don’t thing that using the custom field for this would be clean.

**Reply by desh2608** [reply]
Cool, thanks!
